File: DSA Practice/LeetCode/347. Top K Frequent Elements/solution.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        m = {}
        res = []
        for i in nums:
            if i in m.keys():
                m[i] += 1
            else :
                m[i] = 1

        rank = sorted(list(m.values()), reverse = True)[:k]
        #print(rank)

        for x, v in m.items():
            if v in rank:
                res.append(x)

        return res

File: DSA Practice/LeetCode/347. Top K Frequent Elements/solution.java
import java.util.*;

class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> m = new HashMap<Integer, Integer>();
        int[] res = new int[k];

        for (int i : nums){
            if (m.containsKey(i)){
                m.put(i, m.get(i)+1);
            } else {
                m.put(i, 1);
            }
        }

        // Sorting to get rank
        Collection<Integer> valu = m.values();
        ArrayList<Integer> rank = new ArrayList<Integer>(valu);
        Collections.sort(rank, Collections.reverseOrder());
        
        List needed = rank.subList(0, k);
        int pointer = 0;

        for (Map.Entry e : m.entrySet()){
            int val = (int) e.getValue();
            int key = (int) e.getKey();

            if (needed.contains(val)){
                res[pointer] = key;
                pointer++;
            }
        }

        return res;

    }
}